### PR TITLE
Dbeaver/dbeaver vscode#22 split core plugins

### PR DESCRIPTION
**Pull request by alexander-skoblikov**

### CODE REVIEW

File: server/bundles/io.cloudbeaver.model/src/io/cloudbeaver/model/rm/RMControllerInvocationHandler.java [hunk 1]
```diff
@@ -25,16 +25,16 @@
 import java.lang.reflect.Method;
 
 public class RMControllerInvocationHandler implements InvocationHandler {
```
Comment (yagudin10) [inline server/bundles/io.cloudbeaver.model/src/io/cloudbeaver/model/rm/RMControllerInvocationHandler.java]: looks like this class is not used
we can remove it

File: server/bundles/io.cloudbeaver.model/src/io/cloudbeaver/model/app/BaseServletApplication.java [hunk 2]
```diff
@@ -47,14 +47,14 @@
 /**
  * Web application
```
Comment (yagudin10) [inline server/bundles/io.cloudbeaver.model/src/io/cloudbeaver/model/app/BaseServletApplication.java]: Servlet application

File: server/bundles/io.cloudbeaver.model/src/io/cloudbeaver/model/app/BaseServerConfigurationController.java [hunk 3]
```diff
@@ -43,7 +43,7 @@ public abstract class BaseServerConfigurationController<T extends WebServerConfi
     protected BaseServerConfigurationController(@NotNull Path homeDirectory) {
         this.homeDirectory = homeDirectory;
         //default workspaceLocation
-        this.workspacePath = homeDirectory.resolve("workspace");
+        this.workspacePath = null;
```
Comment (DenisSinelnikov) [inline server/bundles/io.cloudbeaver.model/src/io/cloudbeaver/model/app/BaseServerConfigurationController.java]: why do you remove here?

File: server/bundles/io.cloudbeaver.model/src/io/cloudbeaver/model/app/WebAppConfiguration.java [hunk 4]
```diff
@@ -17,39 +17,38 @@
...
+        return new String[0];
+    }
 
-    boolean isFeaturesEnabled(String[] requiredFeatures);
+    default String[] getEnabledAuthProviders() {
+        return new String[0];
+    }
 
-    boolean isFeatureEnabled(String id);
+    @NotNull
```
Comment (yagudin10) [inline server/bundles/io.cloudbeaver.model/src/io/cloudbeaver/model/app/WebAppConfiguration.java:47]: please annotate methods in children classes

File: server/bundles/io.cloudbeaver.server/src/io/cloudbeaver/service/core/WebServiceBindingCore.java [hunk 5]
```diff
@@ -48,7 +49,7 @@ public WebServiceBindingCore() {
 
     @Override
     public void bindWiring(DBWBindingContext model) throws DBWebException {
-        CBPlatform platform = CBPlatform.getInstance();
+        BaseWebPlatform platform = WebAppUtils.getWebPlatform();
         var sessionManager = platform.getSessionManager();
```
Comment (yagudin10) [inline server/bundles/io.cloudbeaver.server/src/io/cloudbeaver/service/core/WebServiceBindingCore.java]: can be changed to WebAppUtils.getWebApplication().getSessionManager()

File: server/bundles/io.cloudbeaver.server.ce/src/io/cloudbeaver/server/CBPlatform.java [hunk 6]
```diff
@@ -124,7 +120,7 @@ public synchronized void dispose() {
 
     @NotNull
     @Override
-    public GQLApplicationAdapter getApplication() {
+    public WebApplication getApplication() {
```
Comment (yagudin10) [inline server/bundles/io.cloudbeaver.server.ce/src/io/cloudbeaver/server/CBPlatform.java]: we can return CBApplication<?> there

**Review by DenisSinelnikov — APPROVED** [review]

**Pull request merged**